monitoring.tf

```diff
@@ -0,0 +1,174 @@
+# Base infra logging
+# Deploy fluentbit with the fluent-operator and configure it so that pods with the `dai/logging: enable` annotation have
+# they logs pushed to CloudWatch.
+# By default (hardcoded), fluent-operator and fluent-bit will have this annotation set
+locals {
+  # Namespace for the resources deployed by the fluent-operator (fluent-bit will be here too)
+  fluent_namespace                       = "monitoring"
+  fluentbit_cloudwatch_log_group         = "/${local.stack_name}-fluentbit-operator"
+  fluentbit_cloudwatch_log_stream_prefix = "fluentbit-"
+  fluentbit_dai_tag                      = "dai"
+}
+
+# Fluent-operator (https://github.com/fluent/fluent-operator operator)
+resource "helm_release" "fluent_operator" {
+  chart      = "fluent-operator"
+  name       = "fluent-operator"
+  repository = "https://fluent.github.io/helm-charts"
+  version    = "v3.0.0" # Note: using "v3.0" will issue in resource update on each terraform plan/apply
+
+  create_namespace = true
+  namespace        = local.fluent_namespace
+
+  values = [
+    <<-YAML
+    containerRuntime: containerd
+    operator:
+      annotations:
+        dai/logging: enable
+    fluentbit:
+      affinity:
+        nodeAffinity:
+          # Need to be modified because by default fluent-bit will try to start on faregate nodes and it's not working
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/edge # This expression is here by default
+                    operator: DoesNotExist
+                  - key: eks.amazonaws.com/compute-type
+                    operator: NotIn
+                    values:
+                      - fargate
+      annotations:
+        dai/logging: enable
+      # Because we deploy our own pipeline, disable the default one
+      filter:
+        kubernetes:
+          enable: false # true = default
+          annotations: true
+        containerd:
+          enable: true # If disabled, fluent-operator has reconilier error
```

Comment:
```suggestion
          enable: true # If disabled, fluent-operator has reconciler error
```

Reply:
@sgametrio @rbapst-tamedia 

exclude logs annotation is already a built in feature. 

https://docs.fluentbit.io/manual/pipeline/filters/kubernetes#request-to-exclude-logs

Required this to be enabled `K8S-Logging.Exclude` in the config

```
apiVersion: v1
kind: Pod
metadata:
  name: apache-logs
  labels:
    app: apache-logs
  annotations:
    fluentbit.io/exclude: "true"
spec:
  containers:
  - name: apache
    image: edsiper/apache_logs
  ```
  
  One of the main objectives is to have 'infra logs' which are not mixed up with app logs.
  
  I would go the route of that we as the infra managers, annotate all infra related services and we can always log if they have this. This actually allows for many use cases in the future (cost consumption, alerting routing, etc)

Reply:
@sgametrio yes, the difficulty is to define what is an infra or platform application. 
- Log if pod is in a specific namespace (was in a previous commit in this branch). Could probably be extended by filtering on namespace + pod name or pattern.
- Log based on annotation
The problem is that if we don't filter, all logs (for all pods, being infra of app one) will go in cloudwatch and we probably don't want this. But I agree that with the annotation, it could be difficult to log the addons for example.